ingest/anndata_.py

```diff
@@ -35,3 +44,65 @@ def validate(self):
         except ValueError:
             return False
 
+    @staticmethod
+    def generate_cluster_header(adata, clustering_name):
+        """
+        Based on clustering dimensions, write clustering NAME line to file
+        """
+        dim2 = ['NAME', 'X', 'Y']
+        dim3 = ['NAME', 'X', 'Y', 'Z']
+        if adata.obsm[clustering_name].shape[1] == 2:
+            headers = dim2
+        elif adata.obsm[clustering_name].shape[1] == 3:
+            headers = dim3
+        elif adata.obsm[clustering_name].shape[1] > 3:
+            msg = f"Data in obsm \"{clustering_name}\" has too many dimensions for visualization."
+            # !!! is ValueError the right type?
+            raise ValueError(msg)
```

Comment:
I think you're right!  Per docs, [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError) is:
> Raised when an operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as [IndexError](https://docs.python.org/3/library/exceptions.html#IndexError).
```suggestion
            raise ValueError(msg)
```

Reply:
`9bc7f61`